### ISSUE TITLE
enable_history_search conflicts with completion

### ISSUE THREAD

**Issue by mixmastamyk** [jonathanslenders/python-prompt-toolkit#223]
Hi, when I enable this feature, it works, but then my completer stops working.  I'd like to use both of them:

    enable_history_search=True,

I'm using the completer from:

    /examples/regular-language.py

If you happen to read this, is there an example to do a non-regular language? ;)

**Comment by mixmastamyk** [comment]
Btw, latest pptk on Windows 7 and Linux (Wily).

**Comment by ahundt** [comment]
this would also be a great 2.0 feature :-)

**Comment by jonathanslenders** [comment]
Yes, I'm not sure what's the right way to both support history search and the up/down arrows for completion. Maybe ``Ctrl-N`` and ``Ctrl-P`` for completions (which already works), and arrows for history search.

Non-regular language support won't be added as part of prompt_toolkit itself. Even the regular-language module comes with some performance pitfalls in some edge cases making me wonder whether it should stay part of prompt_toolkit itself.

**Comment by laixintao** [comment]
Hi @jonathanslenders 

Can I check if there are any completions on the current cursor position?

I'm thinking using the arrow keys for completions only when there are completions available, and using arrow keys for history search when there are no compiletions.

**Comment by laixintao** [comment]
> Hi @jonathanslenders
> 
> Can I check if there are any completions on the current cursor position?
> 
> I'm thinking using the arrow keys for completions only when there are completions available, and using arrow keys for history search when there are no compiletions.

I just found that I can use `ctrl` + `P` for part history suggestion, forgive the last comment.

**Comment by ubershmekel** [comment]
I would suggest that if the completion menu is open, users can use the up and down keys to browse it. At the top most and bottom most proposed completion the menu does not wrap around but instead closes and goes through history.

The common case is the user hitting "up" for the recent example in history. That would work. And the user can hit "down" to go into the completion menu.

The downside (hah!) is that if you want to go to the future option in history, you have to hit the "down" arrow many times, or some other way dismiss the autocompletion menu. And that you lose the wrapping behavior in the completion menu.
